src/libstore/serve-protocol.hh

```diff
@@ -6,6 +6,9 @@ namespace nix {
 #define SERVE_MAGIC_2 0x5452eecb
 
 #define SERVE_PROTOCOL_VERSION (2 << 8 | 7)
+#define PROTOCOL_MAJOR(x) (((x) & 0xff00) >> 8)
+#define PROTOCOL_MINOR(x) ((x) & 0x00ff)
```

Comment:
Does anything speak against implementing that as a nicely typed function like this:

```cpp
constexpr size_t protocol_minor(size_t x) { return x & 0xff; }
```

This way this happens at compile time if possible and no one will ever have to handle problems that occur from entering the wrong signed/unsigned type.